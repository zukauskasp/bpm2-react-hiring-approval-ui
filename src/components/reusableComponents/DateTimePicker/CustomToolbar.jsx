import React,{useState} from 'react';

import PickerToolbar from "@material-ui/pickers/_shared/PickerToolbar";
import TextField from "../TextField/TextField";

const CustomToolbar = ({selectedDate, isHoursDisabled}) =>{

	const minutes = selectedDate.getMinutes() < 10 ? "0" + selectedDate.getMinutes() : selectedDate.getMinutes();
	const hours = selectedDate.getHours() < 10 ? "0" + selectedDate.getHours() : selectedDate.getHours();

	const [selectedHours, setSelectedHours] = useState(hours + ":" + minutes);

	const onHoursChange = (time) => {
		if (time !== "") {

			var hours = time.split(":")[0];
			var minutes = time.split(":")[1];

			const displayTime = hours + ":" + minutes;

			setSelectedHours(displayTime);
			selectedDate.setHours(hours,minutes)
		}
	}

  return(
    <PickerToolbar title='Select time' isLandscape={true}>
      <	TextField
				name="posponeHours"
				value={selectedHours}
				type="time"
				onChange={(e)=>{onHoursChange(e.target.value)}}
				placeholder=""
				label="Select time"
				readonly={isHoursDisabled}
			/>
    </PickerToolbar>
  )
}

export default CustomToolbar;