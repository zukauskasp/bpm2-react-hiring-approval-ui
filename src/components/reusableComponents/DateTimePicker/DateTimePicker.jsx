import React from 'react';
import PropTypes from 'prop-types';

import DateFnsUtils from '@date-io/date-fns';
import {
  MuiPickersUtilsProvider,
  DatePicker 
} from '@material-ui/pickers';
import { createMuiTheme } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/styles";

import CustomToolbar from "./CustomToolbar";
import Holidays from "date-holidays";

const DateTimePicker = ({country,isWeekendsDisabled,isHolidaysDisabled,selectedDate, setDate,defaultHours,isHoursDisabled, ...otherProps}) => {

    const holidays = isHolidaysDisabled && country ? new Holidays(country.toUpperCase()): null;

    const disableWeekends = (date) => {
      return date.getDay() === 0 || date.getDay() === 6;
    }

    const disableHolidays = (date) => {
      return holidays.isHoliday(date) || disableWeekends(date);
    }
    
    const disableDates = (date) => {
        return ( isWeekendsDisabled && disableWeekends(date)) || (isHolidaysDisabled && disableHolidays(date));
	}

    return (
        <div>
            <MuiPickersUtilsProvider utils={DateFnsUtils}>
            	<ThemeProvider theme={materialTheme}>
					<DatePicker 
						{...otherProps}
						variant="dialog"
						value={selectedDate}
						orientation='landscape'
						ToolbarComponent={()=>(<CustomToolbar selectedDate={selectedDate} isHoursDisabled={isHoursDisabled}/>)}
						onChange={(date) => {setDate(date)}}
						onAccept={(date)=> setDate(selectedDate && date ? 
													new Date(
														date.getFullYear(),
														date.getMonth(),
														date.getDate(),
														selectedDate.getHours(),
														selectedDate.getMinutes()) 
													: date)}

						shouldDisableDate={disableDates}
					/>
                </ThemeProvider>
            </MuiPickersUtilsProvider>
        </div>
    )
}


const materialTheme = createMuiTheme({
	overrides: {
		MuiOutlinedInput: {
			"root": {
				'&$focused $notchedOutline': {
					borderColor: "rgb(0, 159, 218)",
					borderWidth: "1px",
					outline: "0 none"
				},
				"&$error $notchedOutline": {
					borderColor: "rgba(0, 0, 0, 0.23)"
				},
			}
		},
		MuiFormLabel: {
			root: {
				"&$error": {
					color: 'rgba(0, 0, 0, 0.54)'
				}
			}
		},
	  	MuiPickersDay: {
			day: {
		  	color: '#003755'
		},
		daySelected: {
		  	backgroundColor: '#009FDA',
		  		'&:hover': {
					backgroundColor: '#003755'
		  		}
			},
			dayDisabled: {
		  		color: '#C5C8CB',
			},
	  	},
	  	MuiPickersToolbar: {
			toolbar: {
		  		backgroundColor: '#f9fafa',
			},
			toolbarLandscape:{
				alignItems: 'flex-start'
			}
		},
	  	MuiPickersModal: {
			withAdditionalAction:{
		  		backgroundColor: '#f9fafa',
			}
	  	},
	 	MuiButton: {
			textPrimary: {
		  		color:'#009FDA'
	  		}
		},
		MuiTextField: {
			root: {
				width: '280px',
				margin: '0 0 10px 0',
				backgroundColor: '#fff'
			}
		},
		MuiInputBase: {
			root: {
				height: '48px'
			}
		},
		MuiFormHelperText: {
			root: {
				background: '#FBB273',
				position:'relative',
				left: '0',
				top: '110%',
				color: 'white',
				height: '48px',
				lineHeight: '48px',
				fontSize: '14px',
				minWidth: '280px',
				padding: '0 0.8rem',
				marginTop: '10px',

				'&::before': {
					content: '""',
					position: 'absolute',
					top: '-18%',
					width: '0',
					height: '0',
					borderLeft: '10px solid transparent',
					borderRight: '10px solid transparent',
					borderBottom: '10px solid #FBB273'
				},
				"&$error": {
					color: "white"
				},
			},
			contained: {
				marginLeft: 'unset'
			},
			},
	},
});  

DateTimePicker.propTypes = {
	id: PropTypes.string.isRequired,
	label: PropTypes.string.isRequired,
	setDate: PropTypes.func.isRequired,
	selectedDate: PropTypes.instanceOf(Date).isRequired, 
	country: PropTypes.string, 
	defaultHours: PropTypes.string,
	format: PropTypes.string,
	showTodayButton: PropTypes.bool,
	isWeekendsDisabled: PropTypes.bool,
	isHolidaysDisabled: PropTypes.bool,
	isHoursDisabled: PropTypes.bool
} 

DateTimePicker.defaultProps = {
	format: "MM/dd/yyyy HH:mm",
	label: "Select due date",
	showTodayButton: true,
	isWeekendsDisabled: false,
	isHolidaysDisabled: false,
	isHoursDisabled:false,
}

export default DateTimePicker;
