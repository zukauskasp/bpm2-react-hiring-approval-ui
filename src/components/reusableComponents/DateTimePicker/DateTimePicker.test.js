import React from 'react';
import {shallow,mount} from 'enzyme';
import DateTimePicker from './DateTimePicker';

describe('DateTimePicker', ()=> {

    afterAll(() => {
        mount.cleanUp();
      });
      
    it('expect to match latest snapshot', () => {
        const wrapper =  mount(<DateTimePicker id="dateTimePickerId" selectedDate={new Date(1650000000000)} setDate={()=>{}}/>);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('expect to render DateTimePicker', () => {
        expect(shallow(<DateTimePicker id="dateTimePickerId" selectedDate={new Date()} setDate={()=>{}}/>).length).toEqual(1);
    }); 

    it('expect DateTimePicker to have custom date format', () => {
        const wrapper = mount(<DateTimePicker id="dateTimePickerId" selectedDate={new Date(1650000000000)} format={'MM.dd/yyyy HH.mm'} setDate={()=>{}}/>);
        expect(wrapper.props()).toHaveProperty('format', 'MM.dd/yyyy HH.mm');
    }); 

}); 