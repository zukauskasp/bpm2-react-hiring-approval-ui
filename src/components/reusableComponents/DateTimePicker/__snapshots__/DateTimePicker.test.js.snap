// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker expect to match latest snapshot 1`] = `
"<DateTimePicker id=\\"dateTimePickerId\\" selectedDate={{...}} setDate={[Function: setDate]} format=\\"MM/dd/yyyy HH:mm\\" label=\\"Select due date\\" showTodayButton={true} isWeekendsDisabled={false} isHolidaysDisabled={false} isHoursDisabled={false}>
  <div>
    <MuiPickersUtilsProvider utils={[Function: DateFnsUtils]}>
      <ThemeProvider theme={{...}}>
        <PickerWithState id=\\"dateTimePickerId\\" format=\\"MM/dd/yyyy HH:mm\\" label=\\"Select due date\\" showTodayButton={true} variant=\\"dialog\\" value={{...}} orientation=\\"landscape\\" ToolbarComponent={[Function: ToolbarComponent]} onChange={[Function: onChange]} onAccept={[Function: onAccept]} shouldDisableDate={[Function: disableDates]} minDate={{...}} maxDate={{...}} invalidDateMessage=\\"Invalid Date Format\\" minDateMessage=\\"Date should not be before minimal date\\" maxDateMessage=\\"Date should not be after maximal date\\" allowKeyboardControl={true} openTo=\\"date\\" views={{...}}>
          <Wrapper variant=\\"dialog\\" InputComponent={[Function: PureDateInput]} DateInputProps={{...}} format=\\"MM/dd/yyyy HH:mm\\" open={false} onClear={[Function: onClear]} onAccept={[Function: onAccept]} onSetToday={[Function: onSetToday]} onDismiss={[Function: onDismiss]} id=\\"dateTimePickerId\\" label=\\"Select due date\\" showTodayButton={true}>
            <ModalWrapper InputComponent={[Function: PureDateInput]} DateInputProps={{...}} format=\\"MM/dd/yyyy HH:mm\\" open={false} onClear={[Function: onClear]} onAccept={[Function: onAccept]} onSetToday={[Function: onSetToday]} onDismiss={[Function: onDismiss]} id=\\"dateTimePickerId\\" label=\\"Select due date\\" showTodayButton={true} okLabel=\\"OK\\" cancelLabel=\\"Cancel\\" clearLabel=\\"Clear\\" todayLabel=\\"Today\\" clearable={false}>
              <PureDateInput format=\\"MM/dd/yyyy HH:mm\\" id=\\"dateTimePickerId\\" label=\\"Select due date\\" inputValue=\\"04/15/2022 08:20\\" validationError=\\"\\" openPicker={[Function: openPicker]}>
                <WithStyles(ForwardRef(TextField)) error={false} helperText=\\"\\" format=\\"MM/dd/yyyy HH:mm\\" id=\\"dateTimePickerId\\" label=\\"Select due date\\" onClick={[Function: openPicker]} value=\\"04/15/2022 08:20\\" variant={[undefined]} InputProps={{...}} onKeyDown={[Function: onKeyDown]}>
                  <ForwardRef(TextField) classes={{...}} error={false} helperText=\\"\\" format=\\"MM/dd/yyyy HH:mm\\" id=\\"dateTimePickerId\\" label=\\"Select due date\\" onClick={[Function: openPicker]} value=\\"04/15/2022 08:20\\" variant={[undefined]} InputProps={{...}} onKeyDown={[Function: onKeyDown]}>
                    <WithStyles(ForwardRef(FormControl)) className=\\"MuiTextField-root\\" disabled={false} error={false} fullWidth={false} hiddenLabel={[undefined]} required={false} color=\\"primary\\" variant=\\"standard\\" format=\\"MM/dd/yyyy HH:mm\\" onClick={[Function: openPicker]} onKeyDown={[Function: onKeyDown]}>
                      <ForwardRef(FormControl) classes={{...}} className=\\"MuiTextField-root\\" disabled={false} error={false} fullWidth={false} hiddenLabel={[undefined]} required={false} color=\\"primary\\" variant=\\"standard\\" format=\\"MM/dd/yyyy HH:mm\\" onClick={[Function: openPicker]} onKeyDown={[Function: onKeyDown]}>
                        <div className=\\"MuiFormControl-root MuiTextField-root\\" format=\\"MM/dd/yyyy HH:mm\\" onClick={[Function: openPicker]} onKeyDown={[Function: onKeyDown]}>
                          <WithStyles(ForwardRef(InputLabel)) htmlFor=\\"dateTimePickerId\\" id=\\"dateTimePickerId-label\\">
                            <ForwardRef(InputLabel) classes={{...}} htmlFor=\\"dateTimePickerId\\" id=\\"dateTimePickerId-label\\">
                              <WithStyles(ForwardRef(FormLabel)) data-shrink={true} className=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink\\" classes={{...}} htmlFor=\\"dateTimePickerId\\" id=\\"dateTimePickerId-label\\">
                                <ForwardRef(FormLabel) classes={{...}} data-shrink={true} className=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink\\" htmlFor=\\"dateTimePickerId\\" id=\\"dateTimePickerId-label\\">
                                  <label className=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink={true} htmlFor=\\"dateTimePickerId\\" id=\\"dateTimePickerId-label\\">
                                    Select due date
                                  </label>
                                </ForwardRef(FormLabel)>
                              </WithStyles(ForwardRef(FormLabel))>
                            </ForwardRef(InputLabel)>
                          </WithStyles(ForwardRef(InputLabel))>
                          <WithStyles(ForwardRef(Input)) aria-describedby={[undefined]} autoComplete={[undefined]} autoFocus={false} defaultValue={[undefined]} fullWidth={false} multiline={false} name={[undefined]} rows={[undefined]} rowsMax={[undefined]} type={[undefined]} value=\\"04/15/2022 08:20\\" id=\\"dateTimePickerId\\" inputRef={[undefined]} onBlur={[undefined]} onChange={[undefined]} onFocus={[undefined]} placeholder={[undefined]} inputProps={[undefined]} readOnly={true}>
                            <ForwardRef(Input) classes={{...}} aria-describedby={[undefined]} autoComplete={[undefined]} autoFocus={false} defaultValue={[undefined]} fullWidth={false} multiline={false} name={[undefined]} rows={[undefined]} rowsMax={[undefined]} type={[undefined]} value=\\"04/15/2022 08:20\\" id=\\"dateTimePickerId\\" inputRef={[undefined]} onBlur={[undefined]} onChange={[undefined]} onFocus={[undefined]} placeholder={[undefined]} inputProps={[undefined]} readOnly={true}>
                              <WithStyles(ForwardRef(InputBase)) classes={{...}} fullWidth={false} inputComponent=\\"input\\" multiline={false} type=\\"text\\" aria-describedby={[undefined]} autoComplete={[undefined]} autoFocus={false} defaultValue={[undefined]} name={[undefined]} rows={[undefined]} rowsMax={[undefined]} value=\\"04/15/2022 08:20\\" id=\\"dateTimePickerId\\" inputRef={[undefined]} onBlur={[undefined]} onChange={[undefined]} onFocus={[undefined]} placeholder={[undefined]} inputProps={[undefined]} readOnly={true}>
                                <ForwardRef(InputBase) classes={{...}} fullWidth={false} inputComponent=\\"input\\" multiline={false} type=\\"text\\" aria-describedby={[undefined]} autoComplete={[undefined]} autoFocus={false} defaultValue={[undefined]} name={[undefined]} rows={[undefined]} rowsMax={[undefined]} value=\\"04/15/2022 08:20\\" id=\\"dateTimePickerId\\" inputRef={[undefined]} onBlur={[undefined]} onChange={[undefined]} onFocus={[undefined]} placeholder={[undefined]} inputProps={[undefined]} readOnly={true}>
                                  <div className=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" onClick={[Function: handleClick]}>
                                    <input aria-invalid={false} aria-describedby={[undefined]} autoComplete={[undefined]} autoFocus={false} defaultValue={[undefined]} disabled={false} id=\\"dateTimePickerId\\" onAnimationStart={[Function: handleAutoFill]} name={[undefined]} placeholder={[undefined]} readOnly={true} required={false} rows={[undefined]} value=\\"04/15/2022 08:20\\" onKeyDown={[undefined]} onKeyUp={[undefined]} type=\\"text\\" className=\\"MuiInputBase-input MuiInput-input\\" onBlur={[Function: handleBlur]} onChange={[Function: handleChange]} onFocus={[Function: handleFocus]} />
                                  </div>
                                </ForwardRef(InputBase)>
                              </WithStyles(ForwardRef(InputBase))>
                            </ForwardRef(Input)>
                          </WithStyles(ForwardRef(Input))>
                        </div>
                      </ForwardRef(FormControl)>
                    </WithStyles(ForwardRef(FormControl))>
                  </ForwardRef(TextField)>
                </WithStyles(ForwardRef(TextField))>
              </PureDateInput>
              <WithStyles(ModalDialog) wider={[undefined]} showTabs={[undefined]} open={false} onClear={[Function: onClear]} onAccept={[Function: onAccept]} onDismiss={[Function: onDismiss]} onSetToday={[Function: onSetToday]} clearLabel=\\"Clear\\" todayLabel=\\"Today\\" okLabel=\\"OK\\" cancelLabel=\\"Cancel\\" clearable={false} showTodayButton={true}>
                <ModalDialog classes={{...}} wider={[undefined]} showTabs={[undefined]} open={false} onClear={[Function: onClear]} onAccept={[Function: onAccept]} onDismiss={[Function: onDismiss]} onSetToday={[Function: onSetToday]} clearLabel=\\"Clear\\" todayLabel=\\"Today\\" okLabel=\\"OK\\" cancelLabel=\\"Cancel\\" clearable={false} showTodayButton={true}>
                  <WithStyles(ForwardRef(Dialog)) role=\\"dialog\\" onClose={[Function: onDismiss]} classes={{...}} open={false}>
                    <ForwardRef(Dialog) classes={{...}} role=\\"dialog\\" onClose={[Function: onDismiss]} open={false}>
                      <ForwardRef(Modal) className=\\"MuiDialog-root\\" BackdropComponent={{...}} BackdropProps={{...}} closeAfterTransition={true} disableBackdropClick={false} disableEscapeKeyDown={false} onEscapeKeyDown={[undefined]} onClose={[Function: onDismiss]} open={false} role=\\"dialog\\" />
                    </ForwardRef(Dialog)>
                  </WithStyles(ForwardRef(Dialog))>
                </ModalDialog>
              </WithStyles(ModalDialog)>
            </ModalWrapper>
          </Wrapper>
        </PickerWithState>
      </ThemeProvider>
    </MuiPickersUtilsProvider>
  </div>
</DateTimePicker>"
`;
